jupyter_server/services/contents/filemanager.py

```diff
@@ -256,10 +256,28 @@ def _dir_model(self, path, content=True):
         model = self._base_model(path)
         model['type'] = 'directory'
         model['size'] = None
+        MAX_NUM_CONTENTS_PAGE = 100 # A magic number of items that is acceptable to run os.lstat with, and also acceptable to render in the browser
         if content:
             model['content'] = contents = []
             os_dir = self._get_os_path(path)
-            for name in os.listdir(os_dir):
+            dir_contents = os.listdir(os_dir)
+            len_contents = len(dir_contents) 
+            # TODO: some items might be broken symlinks, hidden files, etc. so the real output of each page may have different length
+            
+            if len_contents > MAX_NUM_CONTENTS_PAGE:
+                # round up the max number of pages
+                max_page = int(len_contents / MAX_NUM_CONTENTS_PAGE) + (len_contents % MAX_NUM_CONTENTS_PAGE > 0)
```

Comment:
there'a handy function for calculating div and mod. 
```python
pages, contents_remainder = divmod(len_contents, MAX_NUM_CONTENTS_PAGE)
max_page = pages + contents_remainder
```

Reply:
Thanks for the divmod function! The value of `max_page` should be the same as `math.ceil(len_contents / MAX_NUM_CONTENTS_PAGE)`, so `max_page =  pages + contents_remainder` isn't correct actually.